libraries/chain/transaction_context.cpp

```diff
@@ -496,17 +496,26 @@ namespace eosio { namespace chain {
             } else {
                // exceeds trx.max_cpu_usage_ms or cfg.max_transaction_cpu_usage if objective_duration_limit is greater
                const int64_t cpu_limit = (cpu_limited_by_account ? account_cpu_limit : objective_duration_limit.count());
-               EOS_ASSERT( billed_us <= cpu_limit,
-                           tx_cpu_usage_exceeded,
-                           "billed CPU time (${billed} us) is greater than the maximum billable CPU time for the transaction (${billable} us)",
-                           ("billed", billed_us)( "billable", cpu_limit )
-               );
+               if( subjective_billed_us > 0 ) {
+                    EOS_ASSERT( billed_us <= cpu_limit,
+                              tx_cpu_usage_exceeded,
+                              "billed CPU time (${billed} us) is greater than the maximum billable CPU time for the transaction (${billable} us) with a subjective cpu of (${subjective} us)",
+                              ("billed", billed_us)( "billable", cpu_limit )( "subjective", subjective_billed_us )
+                  );
+               }
+               else {
+                  EOS_ASSERT( billed_us <= cpu_limit,
+                              tx_cpu_usage_exceeded,
+                              "billed CPU time (${billed} us) is greater than the maximum billable CPU time for the transaction (${billable} us)",
+                              ("billed", billed_us)( "billable", cpu_limit )
+                  );
+               }
             }
          }
       }
    }
 
```

Comment:
I think we should also update the checktime() messages to include subjective billing as well.

Reply:
done but not sure that the same message add-on should be used here

Reply:
I don't see the checktime() modification.